coalib/bearlib/abstractions/Linter.py

```diff
@@ -368,6 +369,43 @@ def process_output_corrected(self,
                              diffs={filename: diff},
                              severity=diff_severity)
 
+        def process_output_unidiff(self,
+                                   output,
+                                   filename,
+                                   file,
+                                   diff_severity=RESULT_SEVERITY.NORMAL,
+                                   result_message='Inconsistency found.',
+                                   diff_distance=1):
+            """
+            Processes the executable's output as a unified diff.
+
+            :param output:
+                The output of the program as a string.
+            :param filename:
+                The filename of the file currently being corrected.
+            :param file:
+                The contents of the file currently being corrected.
+            :param diff_severity:
+                The severity to use for generating results.
+            :param result_message:
+                The message to use for generating results.
+            :param diff_distance:
+                Number of unchanged lines that are allowed in between two
+                changed lines so they get yielded as one diff. If a negative
+                distance is given, every change will be yielded as an own diff,
+                even if they are right beneath each other.
+            :return:
+                An iterator returning results containing patches for the
+                file to correct.
+            """
+            for diff in Diff.from_unidiff(
+                    output).split_diff(distance=diff_distance):
+                yield Result(self,
+                             result_message,
+                             affected_code=diff.affected_code(filename),
+                             diffs={filename: diff},
+                             severity=diff_severity)
```

Comment:
I got an awesome idea to avoid code duplication:

Create a `process_diff` function, which takes in: `filename`, `diff_severity`, `result_message` and `diff_distance`. It will contain nearly the same code, but:
```
for splitted_diff in diff.split_diff(distance=diff_distance):
    yield Result(...)
```
and `diff` is another parameter.

Then you can refactor the current `process_output_corrected` and `process_output_unidiff`, to just call `process_diff`:
```python
return process_diff(Diff.from_string_arrays(...))
# and
return process_diff(Diff.from_unidiff(...))
```